modules/library/render/src/main/java/org/geotools/renderer/lite/gridcoverage2d/GridCoverageRenderer.java

```diff
@@ -627,8 +631,12 @@ private GridCoverage2D affine(GridCoverage2D input, double[] bkgValues) {
         try {
             ImageWorker iw = new ImageWorker(finalImage);
             iw.setRenderingHints(hints);
+            iw.setROI(roi);
+            iw.setnoData(noData);
             iw.affine(finalRasterTransformation, interpolation, bkgValues);
             im = iw.getRenderedImage();
+            roi = iw.getROI();
+            noData = iw.getNoData();
```

Comment:
Question, so roi and nodata gets updated and now represent those of the transformed image?